slides/week2/week2.org

```diff
@@ -0,0 +1,82 @@
+#+TITLE: Week 2
+#+AUTHOR: Kristaps Berzinch, Raymond Ortiz
+#+EMAIL: kristaps@robojackets.org, rortiz9@gatech.edu
+
+* What are we doing today?
+- Introduction to C++
+ - What it is
+ - Functions
+ - Includes and headers
+ - Working with strings
+- How to find and read documentation
+- Sending and recieving changes from remote Git repositories
+
+* What is C++?
+- A general-purpose programming language
+- Designed with performance, efficiency, and flexibility in mind
+- Great for robots!
+
+* Functions
+#+BEGIN_NOTES
+After introducing the concept, write a simple example. Don't go into prototypes yet, we'll cover them in a bit.
+
+TODO I think we should design a specific example that is easy to code on the fly but isn't completely trivial (e.g. writing an adder function seems dumb)
+#+END_NOTES
+- A function is a chunk of code with defined inputs and outputs
+ - Minimizes copy-and-paste
+ - Less buggy
+- These can be chained together for more complex functionality
+- You want each function to have a single high-level purpose
+
+* Includes
+#+BEGIN_NOTES
+After introducing the concept, demo some basic examples of including a file within another file. Still no prototypes.
+Include a function declaration in more than one file to cause a compilation error to lead into header files
+#+END_NOTES
+- Allows you to use code from other files
+- C++ has a lot of flexibility; this is a blessing and a curse
+- Your code will have little structure unless you organize it well
+- Break up your code logically and purposefully
+ - Readability above all

```

Comment:
Maybe note how 'including' literally inserts the file where the include is? I'm not sure if we're trying to hide that from them at this point, but I found includes/headers much easier to understand once I understood this.